examples/jsm/nodes/inputs/Matrix3Node.js

```diff
@@ -3,9 +3,9 @@ import { Matrix3 } from 'three';
 
 class Matrix3Node extends InputNode {
 
-	constructor( value = new Matrix3() ) {
+	constructor( value = new Matrix3(), type = '' ) {
 
-		super( 'mat3' );
+		super( type + 'mat3' );
```

Comment:
I think we can rename it to `componentType`?

Reply:
Can you remove only the `componentType` for the inputs and let's keep this only in `ShaderNode` like a conversion type? 
https://github.com/mrdoob/three.js/pull/23646#issuecomment-1058420778

So we can do a full integration in the future for the inputs

Reply:
I don't think this is possible... `componentType` is used for setting the node type of an input, while `type` in ConvertType is used only for the type of the ConvertNode.

Reply:
Maybe for it you can need to ignore the InputNode like Vector3Node and use only a ConvertNode for theses conversions ignoring any uniform type.

Reply:
I.e. something like this? https://github.com/mrdoob/three.js/pull/23646/commits/2e42c7d026a45117ea60e7c540da3e0dbdee1b8d

Reply:
> I.e. something like this? https://github.com/mrdoob/three.js/commit/2e42c7d026a45117ea60e7c540da3e0dbdee1b8d

Yeah, something like this.

Reply:
It can generate const like `uvec3( 1, 1, 1)` ?

Reply:
Yes - it would be compiled in something like `uvec3( vec3( 1, 1, 1 ) )`.

Reply:
An `InputConstNode` for constants would be nice.. It can be:
```js
Node <- InputConstNode <- InputNode <- Vector3Node
```

And use `InputConstNode` for non uniforms vars. 
This way we would avoid unnecessary conversions.

Reply:
But it can be done in another PR.

Reply:
@sunag I think the best idea would be to split InputNode entirely into ConstNode and UniformNode, and remove FloatNode, IntNode, etc (keeping the TextureNode and BufferNode - TextureNode could be rewrited without the InputNode/ConstNode/UniformNode and BufferNode could be rewritted with the ConstNode or even without it) in favour of `new ConstNode( value, type )` and `new UniformNode( value, type )`.

But this would be a large breaking change for those who use the Nodes system directly (i.e. not via the ShaderNode).

Reply:
I think it's easier to just add a `ConstNode` between `Node` and `InputNode`.  I think that using classes appropriate to objects we have greater controls, such as optional arguments for example, still not implemented like: 

```js
new Nodes.Vector3Node( 1, 1, 1 );
```

But I think we can leave that for the next PR.

Reply:
My approach can also use optional arguments (if `type` would be the first argument). Will try to make a PR soon...